pandas-stubs/core/strings.pyi

```diff
@@ -65,6 +68,10 @@ class StringMethods(NoNewAttributesMixin, Generic[T, _TS, _TM]):
         self, pat: str = ..., *, n: int = ..., expand: Literal[True], regex: bool = ...
     ) -> _TS: ...
     @overload
+    def split(
+        self, pat: str = ..., *, n: int = ..., expand: Literal[False], regex: bool = ...
+    ) -> _TS2: ...
+    @overload
     def split(
         self, pat: str = ..., *, n: int = ..., expand: bool = ..., regex: bool = ...
     ) -> T: ...
```

Comment:
Combine these two, as `expand` has a default value of `False`.  so if you have `expand: Literal[False] = ...` returning `_TS2`, I think it will work right.